include/boost/url/url.hpp

```diff
@@ -119,7 +119,15 @@ class BOOST_SYMBOL_VISIBLE url
         relative-ref  = relative-part [ "?" query ] [ "#" fragment ]
         @endcode
 
-        @throw std::invalid_argument parse error.
+        @par Exception Safety
+        Strong guarantee.
+        Calls to allocate may throw.
+        Exceptions thrown on invalid input.
+
+        @throw system_error
+        The string contains an invalid <em>URI-reference</em>.
```

Comment:
these exceptions aren't correct

Reply:
`grammar::parse(s, uri_reference_rule).value()` can throw `system_error` and `url(url_view_base const& u)` can throw `std::length_error`. Is that incorrect?